### ISSUE TITLE
sync:recipes --force sometimes asks to overwrite uncommitted changes it makes

### ISSUE THREAD

**Issue by weaverryan** [symfony/flex#456]
Hi friends!

I was just testing the new `sync:recipes --force` command - it's very interesting :). There's on issue I noticed. The command:

A) First updated my `phpunit.xml.dist` file to add a bunch of `env` vars I was missing from outdated recipes
B) Then asked me later if I wanted to overwrite this file, because it had uncommitted changes. After saying "yes", it remove all the env vars and updated it to the latest version.

So basically: sometimes it may ask to override uncommitted changes that IT made. Nicolas mentioned this may be due to the relationship between the configurators.

To replicate

```
git clone git@github.com:symfony/symfony-demo.git
cd symfony-demo
git checkout v1.2.7
composer install
rm -rf vendor/symfony/flex
git clone git@github.com:symfony/flex.git vendor/symfony/flex
composer sync-recipes --force
```
